### DISCUSSION TITLE
How to make PulseVolume widget change volume for all devices (including bluetooth speakers)

### DISCUSSION THREAD

**Discussion by ItsSunnyMonster** [qtile/qtile#3559] [Q&A]
The title basically says it all. I noticed that when I connect a bluetooth speaker, scrolling on the widget still changes the default output devices and not the bluetooth speaker. Is there some way I can make it change volume for all devices?
or at the very least change the volume for the output device that is currently playing the audio?

**Comment by SweepingBishops** [comment]
There are a few ways:
- Write a short script that changes the volume as you need it and bind that script to `Button4` and `Button5` in your config for the widget (using the `mouse_callbacks` parameter).
- Edit the widget's code (and maybe open a PR?)
- Write your own widget.

The last option is what I've done. [my_volume_widget](https://github.com/SweepingBishops/qtile/blob/rewrite-development/myWidget/my_volume_widget.py)
It uses pulsemixer to set the volume. It also solves a few other issues I had with the PulseVolume widget.
DISCLAIMER: I make no claims as to the condition of my widget, it was intended for person use only, it may or may not work for you.
